src/main/java/kitEditor/KitEditor.java

```diff
@@ -1132,7 +1140,62 @@ private void trimAllSamples() {
                 "Done",
                 JOptionPane.INFORMATION_MESSAGE);
     }
+
+    private void duplicateSample(Sample sample) {
+        if (sample == null) {
+          return;
+        }
+        int dest = firstFreeSampleSlot();
+        Sample firstSample = sample;
+        Sample dupeSample;
+        if (dest != -1) {
+            // copy sample data
+            try {
+            dupeSample = Sample.dupeSample(firstSample);
+            dupeSample.reload(halfSpeed.isSelected());
+            } catch (Exception e) {
+                showFileErrorMessage(e);
+                return;
+            }
+            samples[selectedBank][dest] = dupeSample;
+            renameSample(dest, firstSample.getName());
+            if (bytesFree() < 0 && firstSample.canAdjustVolume()) {
+                int fixedTrim = dupeSample.getTrim() - bytesFree() / 16;
+                assert fixedTrim > 0;
+                dupeSample.setTrim(fixedTrim);
+            } else if (bytesFree() < 0) {
+                samples[selectedBank][dest] = null;
+                JOptionPane.showMessageDialog(contentPane,
```

Comment:
Maybe in the future, do throw exceptions instead to scope the whole algorithm with one `try/except` that'll pop the error message for all kinds of issues that can raise here up at once?